src/Utils.ts

```diff
@@ -156,6 +156,16 @@ export function getProvider(): providers.Provider {
   );
 }
 
+export const buildServerUrl = () => {
+  const {
+    app: { url, port },
+  } = getServerConfig();
+
+  const portFromUrl = url.match(/:(\d{0,5})$/);
+
+  return !portFromUrl && port ? `${url}:${port}` : url;
```

Comment:
I know you didn't introduce this, but I've just realized a different solution (I'm not a fan of regexp)
```suggestion
  const parsedUrl = new URL(url);
  const portFromUrl = parsedUrl.port;
  
  if (!portFromUrl && port) {
    parsedUrl.port = port;
  }
  
  return parsedUrl.toString():
```

Reply:
The main issue with this approach is that returns the address with the trailing slash, Ill be removing it.

Reply:
I think we solved it by fixing the client also.